File: app/src/main/java/com/ty/zbpet/ui/adapter/diffadapter/SendOutDiffUtil.java
package com.ty.zbpet.ui.adapter.diffadapter;

import android.support.v7.util.DiffUtil;

import com.ty.zbpet.bean.product.ProductDetails;

import java.util.List;

/**
 * @author TY on 2018/12/4.
 * <p>
 * 发货出库 待办详情 添加按钮
 */
public class SendOutDiffUtil extends DiffUtil.Callback {

    private List<ProductDetails.ListBean> oldList, newList;

    public SendOutDiffUtil(List<ProductDetails.ListBean> oldList, List<ProductDetails.ListBean> newList) {
        this.oldList = oldList;
        this.newList = newList;
    }


    @Override
    public int getOldListSize() {
        return oldList == null ? 0 : oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList == null ? 0 : newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {

        return oldList.get(oldItemPosition).getClass().equals(newList.get(newItemPosition).getClass());
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {

        ProductDetails.ListBean oldBean = oldList.get(oldItemPosition);
        ProductDetails.ListBean newBean = newList.get(newItemPosition);

        return oldBean.equals(newBean);
    }
}

File: app/src/main/java/com/ty/zbpet/ui/adapter/diffadapter/SendOutDiffUtil.kt
package com.ty.zbpet.ui.adapter.diffadapter

import android.support.v7.util.DiffUtil

import com.ty.zbpet.bean.product.ProductDetails

/**
 * @author TY on 2018/12/4.
 *
 *
 * 发货出库 待办详情 添加按钮
 */
class SendOutDiffUtil(private val oldList: List<ProductDetails.ListBean>?, private val newList: List<ProductDetails.ListBean>?) : DiffUtil.Callback() {


    override fun getOldListSize(): Int {
        return oldList?.size ?: 0
    }

    override fun getNewListSize(): Int {
        return newList?.size ?: 0
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {

        return oldList!![oldItemPosition].javaClass == newList!![newItemPosition].javaClass
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {

        val oldBean = oldList!![oldItemPosition]
        val newBean = newList!![newItemPosition]

        return oldBean == newBean
    }
}
